addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -236,29 +356,48 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
     /**
      * @override
      */
-    _renderCustomWidgets: async function (uiFragment) {
-        const checkboxes = [];
+    async _renderCustomXML(uiFragment) {
+        const $select = $(uiFragment.querySelector('we-select'));
         return this._rpc({
-            model: 'product.style',
+            model: 'product.ribbon',
             method: 'search_read',
-        }).then(async data => {
-            for (var k in data) {
-                const checkboxWidget = this._registerUserValueWidget('we-checkbox', this, data[k]['name'], {
-                    dataAttributes: {
-                        'style': data[k]['id'],
-                        'selectClass': data[k]['html_class'],
-                    },
-                });
-                checkboxes.push(checkboxWidget);
-                await checkboxWidget.appendTo(document.createDocumentFragment());
-            }
-        }).then(() => {
-            const menuEl = uiFragment.querySelector('[name="style"]');
-            for (const checkbox of checkboxes) {
-                menuEl.appendChild(checkbox.el);
+            fields: ['id', 'name', 'color', 'text_color'],
+        }).then(async ribbons => {
+            this.ribbons = Object.fromEntries(ribbons.map(r => [r.id, r]));
+            for (const ribbon of ribbons) {
+                $select.append($(`<we-button data-set-ribbon="${ribbon.id}">${ribbon.name}</we-button>`));
             }
         });
     },
+    /**
+     * @override
+     */
+    async _computeWidgetState(methodName, params) {
+        switch (methodName) {
+            case 'setRibbon':
+                return this.$target.attr('data-ribbon-id') || '';
+            case 'setRibbonColor':
+                return this.ribbonColor.replace(/^bg-/, '');
+            case 'setRibbonTextColor': {
+                const match = this.ribbonTextColor.match(/^var\(--(.*)\)$/);
+                return match ? match[1] : this.ribbonTextColor;
+            }
+            case 'setRibbonName':
+                return this.ribbonName;
+            default:
+                return this._super(...arguments);
+        }
```

Comment:
I am not a fan of functions which mean to return something but are doing it in the else clause... Just convention but I would return `_super` at the end of the function and not have a `default` clause.